Assets/RedBlueGames/BulkRename/Editor/BulkRenamerWindow.cs

```diff
@@ -23,712 +23,755 @@
+			var selectedNames = this.GetNamesFromObjectsToRename ();
+			var namePreviews = this.bulkRenamer.GetRenamePreviews (selectedNames);
+
+			for (int i = 0; i < namePreviews.Count; ++i) {
+				var info = new PreviewRowInfo ();
+				var namePreview = namePreviews [i];
+				info.OriginalName = namePreview.OriginalName;
+				info.DiffName = namePreview.GetDiffAsFormattedString (AddedTextColorTag, DeletedTextColorTag);
+				info.NewName = namePreview.NewName;
+				info.Icon = GetIconForObject (this.ObjectsToRename [i]);
+
+				previewRowInfos [i] = info;
+			}
+
+			return previewRowInfos;
+		}
+
+		private string[] GetNamesFromObjectsToRename ()
+		{
+			int namesCount = this.ObjectsToRename.Count;
+			var names = new string[namesCount];
+			for (int i = 0; i < namesCount; ++i) {
+				names [i] = this.ObjectsToRename [i].name;
+			}
+
+			return names;
+		}
+
+		private void RenameAssets ()
+		{
+			// Record all the objects to undo stack, though this unfortunately doesn't capture Asset renames
+			Undo.RecordObjects (this.ObjectsToRename.ToArray (), "Bulk Rename");
+
+			var names = this.GetNamesFromObjectsToRename ();
+			var newNames = this.bulkRenamer.GetRenamePreviews (names);
+
+			for (int i = 0; i < newNames.Count; ++i) {
+				var infoString = string.Format (
+					                 "Renaming asset {0} of {1}",
+					                 i,
+					                 newNames.Count);
+
+				EditorUtility.DisplayProgressBar (
+					"Renaming Assets...",
+					infoString,
+					i / (float)newNames.Count);
+
+				this.RenameObject (this.ObjectsToRename [i], newNames [i].NewName);
+			}
+			RenameSprites (allSpritesWithNewNames);
+
+			EditorUtility.ClearProgressBar ();
+			allSpritesWithNewNames.Clear ();
+
+		}
+
+		private void RenameObject (UnityEngine.Object obj, string newName)
+		{
+			if (AssetDatabase.Contains (obj)) {
+				this.RenameAsset (obj, newName);
+			} else {
+				this.RenameGameObject (obj, newName);
+			}
+		}
+
+		private void RenameGameObject (UnityEngine.Object gameObject, string newName)
+		{
+			gameObject.name = newName;
+		}
+
+		public 	static string ReplaceSpritePrefixInMetafile (string metafileText, string prefixToReplace, string newPrefix)
+		{			
+			string fileIDPattern = prefixToReplace;
+			var fileIDRegex = new System.Text.RegularExpressions.Regex (fileIDPattern);
+			string replacementText = newPrefix;
+			return fileIDRegex.Replace (metafileText, replacementText);
+		}
+
+		void RenameSprites (Dictionary<Object,string> allSpritesWithNewNames)
+		{
+			string actualAssetName;
+			string newAssetName;
+			string path;
+			HashSet<string> assetsToReimport = new HashSet<string> ();
+
+			foreach (var asset in allSpritesWithNewNames) {
+				actualAssetName = asset.Key.name;
+				newAssetName = asset.Value;
+				path = AssetDatabase.GetAssetPath (asset.Key);
+				TextureImporter parentAsset = TextureImporter.GetAtPath (path)as TextureImporter;
+				#region Changing name in metafiles
```

Comment:
RBG is just Red Blue Games (our indie game studio). I generally think regions are an indication that a class should be smaller. This class is my largest class in here and it's much too big.